csrc/mmdeploy/core/registry.h

```diff
@@ -3,55 +3,98 @@
 #ifndef MMDEPLOY_REGISTRY_H
 #define MMDEPLOY_REGISTRY_H
 
-#include <iostream>
-#include <map>
+#include <functional>
 #include <memory>
+#include <optional>
 #include <string>
+#include <string_view>
 #include <vector>
 
-#include "macro.h"
-#include "value.h"
+#include "mmdeploy/core/macro.h"
+#include "mmdeploy/core/mpl/span.h"
 
 namespace mmdeploy {
 
-namespace detail {
+template <typename T>
+struct basic_type {
+  using type = T;
+};
+
+template <typename Iterator>
+class iterator_range {
+ public:
+  explicit iterator_range(Iterator first, Iterator last) : first_(first), last_(last) {}
+  explicit iterator_range(std::pair<Iterator, Iterator> range)
+      : iterator_range(range.first, range.second) {}
 
-template <typename EntryType, typename = void>
-struct get_return_type {
-  using type = std::unique_ptr<EntryType>;
+  auto begin() noexcept { return first_; }
+  auto end() noexcept { return last_; }
+
+ private:
+  Iterator first_;
+  Iterator last_;
 };
 
-template <typename EntryType>
-struct get_return_type<EntryType, std::void_t<typename EntryType::type>> {
-  using type = typename EntryType::type;
+namespace _registry {
+
+using std::optional;
+using std::string_view;
+
+template <typename T, typename = void>
+struct _get_signature {
+  static_assert(!std::is_same_v<T, T>, "tag T is not associated with a signature");
```

Comment:
Why not `static_assert(0, ..)` directly?

Reply:
`static_assert(!std::is_same_v<T, T>)` only fails when the template gets instantiated, whereas `static_assert(0)` always fail.